lib/route.js

```diff
@@ -298,7 +298,7 @@ function buildRouting (options) {
           context.schema = normalizeSchema(context.schema)
 
           const schemaController = this[kSchemaController]
-          if (!opts.validatorCompiler) {
+          if (!opts.validatorCompiler && (opts.schema.body || opts.schema.headers || opts.schema.querystring || opts.schema.params)) {
```

Comment:
> Will this conditional fail if the reference is shifted from the response, as in the bug report, to one of these others, e.g. params?

Good catch, a test for this case is failing. I'm still working on it 👍

Reply:
This optimization will reduce the amount of ajv instances built: [here why](https://github.com/fastify/fastify/pull/2925#issuecomment-798769632)

Reply:
Is this conditional still required after the changes to `fast-json-stringify`?

Reply:
It is not required, since the bump of fjs has solved the issue.

Reverting this line would instantiate an unused ajv instance (wasting memory etc..)

Reply:
So is this PR still valid? I'm not clear about the status.

Reply:
Yes, I think it is a good improvement